WordPress/Classes/Stores/RemoteFeatureFlagStore.swift

```diff
@@ -3,23 +3,25 @@ import WordPressKit
 
 class RemoteFeatureFlagStore {
 
-    /// Alllow injecting the FeatureFlagRemote for tests (or to customize the WordPress.com API)
-    private let remote: FeatureFlagRemote
+    public static let shared = RemoteFeatureFlagStore()
```

Comment:
If we don't go with a global state approach, these are the other solutions I can think of are:

1. Create a new instance of `RemoteFeatureFlagStore` whenever we need it. But this will make[`RemoteFeatureFlagStore.queue`](https://github.com/wordpress-mobile/WordPress-iOS/blob/ca43f851ad373c2142531327b4ac37bf0e3f1b7d/WordPress/Classes/Stores/RemoteFeatureFlagStore.swift#L62), which is used for thread safety, redundant. As we'll be accessing the same cache from different instances, hence, different queues.
2. Couple `RemoteFeatureFlagStore` with `WordPressAppDelegate` which was the original solution. However, this comes with a few downsides:
    1. We'll have to either expose the instance held by `WordPressAppDelegate` so it can be used to retrieve flag values from the cache. Or add a public API that wraps around `RemoteFeatureFlagStore.value(for:)` and `RemoteFeatureFlagStore.hasValue(for:)`.
    2. `WordPressAppDelegate.shared` [returns an optional](https://github.com/wordpress-mobile/WordPress-iOS/blob/ca43f851ad373c2142531327b4ac37bf0e3f1b7d/WordPress/Classes/System/WordPressAppDelegate.swift#L85). So every time we retrieve a flag value through `WordPressAppDelegate.shared` that value will be an optional as well. Plus it [can only be accessed from the main thread](https://github.com/wordpress-mobile/WordPress-iOS/blob/ca43f851ad373c2142531327b4ac37bf0e3f1b7d/WordPress/Classes/System/WordPressAppDelegate.swift#L86).
    3. This opens the possibility of API misuse. In the future, someone might decide to create a separate instance from the one in `WordPressAppDelegate` and use it, and this introduces the same problem as solution 1.

Given this, I feel a global singleton would be our best bet. What do you think? And did you have something else in mind?

Reply:
> If we were to implement the cache on a lower level (ie – on a per-key basis instead of reading and writing the entire dictionary each time)

This is a good idea. But other than the performance trade-off, it's also a bigger change and will require more testing.

> look at the DispatchQueue as a dependency – it can be injected too!

Nice! I ended up doing that and ditching the Singleton pattern. Done in b9e77b0ba75fc79acde342ec100913d8737da5af 👍